scipy/stats/_hypotests.py

```diff
@@ -389,3 +391,192 @@ def _get_wilcoxon_distr(n):
+    ``scipy.stats.norm.rvs`` have the same distribution. We choose a
+    significance level of alpha=0.05.
+
+    >>> import numpy as np
+    >>> from scipy import stats
+    >>> np.random.seed(626)
+    >>> x = stats.norm.rvs(size=100)
+    >>> y = stats.norm.rvs(size=70)
+    >>> res = stats.cramervonmises_2samp(x, y)
+    >>> res.statistic, res.pvalue
+    (0.24331932773109344, 0.19815666195647663)
+
+    The p-value exceeds our chosen significance level, so we do not
+    reject the null hypothesis that the observed samples are drawn from the
+    same distribution.
+
+    For small sample sizes, one can compute the exact p-values:
+
+    >>> x = stats.norm.rvs(size=7)
+    >>> y = stats.t.rvs(df=2, size=6)
+    >>> res = stats.cramervonmises_2samp(x, y, method='exact')
+    >>> res.statistic, res.pvalue
+    (0.2655677655677655, 0.18706293706293706)
+
+    The p-value based on the asymptotic distribution is a good approximation
+    even though the sample size is small.
+    
+    >>> res = stats.cramervonmises_2samp(x, y, method='asymptotic')
+    >>> res.statistic, res.pvalue
+    (0.2655677655677655, 0.17974247316290415)
+
+    Independent of the method, one would not reject the null hypothesis at the
+    chosen significance level in this example.
+    """
+
+    xa = np.sort(np.asarray(x))
+    ya = np.sort(np.asarray(y))
+
+    if xa.size <= 1 or ya.size <= 1:
+        raise ValueError('x and y must contain at least two observations.')
+    if xa.ndim > 1 or ya.ndim > 1:
+        raise ValueError('The samples must be one-dimensional.')
+    if method not in ['auto', 'exact', 'asymptotic']:
+        raise ValueError('method must be either auto, exact or asymptotic.')
+
+    nx = len(xa)
+    ny = len(ya)
+
+    if method == 'auto':
+        if max(nx, ny) > 10:
+            method = 'asymptotic'
+        else:
+            method = 'exact'
+
+    # get ranks of x and y in the pooled sample
+    z = np.concatenate([xa, ya])
+    # in case of ties, use midrank (see [1])
+    r = scipy.stats.rankdata(z, method='average')
+    rx = r[:nx]
+    ry = r[nx:]
+
+    # compute U (eq. 10 in [2])
+    u = nx * np.sum((rx - np.arange(1, nx+1))**2)
+    u += ny * np.sum((ry - np.arange(1, ny+1))**2)
+
+    # compute T (eq. 9 in [2])
+    k, N = nx*ny, nx + ny
+    t = u / (k*N) - (4*k - 1)/(6*N)
+
+    if method == 'exact':
+        p = _pval_cvm_2samp_exact(u, nx, ny)
+    else:
+        # compute expected value and variance of T (eq. 11 and 14 in [2])
+        et = (1 + 1/N)/6
+        vt = (N+1) * (4*k*N - 3*(nx**2 + ny**2) - 2*k)
+        vt = vt / (45 * N**2 * 4 * k)
+
+        # computed the normalized statistic (eq. 15 in [2])
+        tn = 1/6 + (t - et) / np.sqrt(45 * vt)
+
+        # approximate distribution of tn with limiting distribution
+        # of the one-sample test statistic
+        # if tn < 0.0035, the _cdf_cvm_inf(tn) < 4.9*1e-16, return 1.0 directly
+        if tn < 0.0035:
```

Comment:
```suggestion
        # if tn < 0.003, 1 -_cdf_cvm_inf(tn) == 1, so return 1.0 directly
        if tn < 0.003:
```
Not that it matters much practically, but the intent is that this number is small enough that
```
1-_cdf_cvm_inf(tn_crit)==1
```
On my machine, `1-_cdf_cvm_inf(0.0034999)` is `0.9999999999999996 != 1`, so this went a little far, I think. I'd suggest settling on 0.003 just to be safe. Sorry to be a pest! I don't care much, but I get the sense that some others do.

Reply:
done!